### ISSUE TITLE
Could not download video on bilibili

### ISSUE THREAD

**Issue by kidozh** [soimort/you-get#2127]
Please make sure these boxes are checked before submitting your issue – thank you!

- [ ] You can actually watch the video in your browser or mobile application, but not download them with `you-get`.
- [ ] Your `you-get` is up-to-date.
- [ ] I have read <https://github.com/soimort/you-get/wiki/FAQ> and tried to do so.
- [ ] The issue is not yet reported on <https://github.com/soimort/you-get/issues> or  <https://github.com/soimort/you-get/wiki/Known-Bugs>. If so, please add your comments under the existing issue.
- [ ] The issue (or question) is really about `you-get`, not about some other code or project.

Run the command with the `--debug` option, and paste the full output inside the fences:

```
➜  ~ you-get 'http://www.bilibili.com/video/av11839449/' --debug -x 192.168.191.1:1080
[DEBUG] get_content: http://www.bilibili.com/video/av11839449/
[DEBUG] get_content: http://interface.bilibili.com/playurl?&cid=19551835&from=miniplay&player=1&sign=755fca43ed73826c9aae7a0fb4ba08de
[DEBUG] url_info: http://ws.acgvideo.com/8/37/19551835-1.flv?wsTime=1499169204&platform=pc&wsSecret2=9761c1664ee009c230e12eea8628a517&oi=1760342897&rate=6
you-get: version 0.4.750, a tiny downloader that scrapes the web.
you-get: ['http://www.bilibili.com/video/av11839449/']
Traceback (most recent call last):
  File "/usr/local/bin/you-get", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.5/dist-packages/you_get/__main__.py", line 92, in main
    main(**kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 1444, in main
    script_main('you-get', any_download, any_download_playlist, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 1355, in script_main
    download_main(download, download_playlist, args, playlist, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 1157, in download_main
    download(url, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 1437, in any_download
    m.download(url, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/extractors/bilibili.py", line 187, in bilibili_download
    info_only=info_only)
  File "/usr/local/lib/python3.5/dist-packages/you_get/extractors/bilibili.py", line 101, in bilibili_download_by_cid
    _, type_, temp = url_info(url, headers={'Referer': 'http://www.bilibili.com/'})
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 420, in url_info
    response = urlopen_with_retry(request.Request(url, headers=headers))
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 307, in urlopen_with_retry
    return request.urlopen(*args, **kwargs)
  File "/usr/lib/python3.5/urllib/request.py", line 163, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.5/urllib/request.py", line 472, in open
    response = meth(req, response)
  File "/usr/lib/python3.5/urllib/request.py", line 582, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python3.5/urllib/request.py", line 504, in error
    result = self._call_chain(*args)
  File "/usr/lib/python3.5/urllib/request.py", line 444, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.5/urllib/request.py", line 696, in http_error_302
    return self.parent.open(new, timeout=req.timeout)
  File "/usr/lib/python3.5/urllib/request.py", line 472, in open
    response = meth(req, response)
  File "/usr/lib/python3.5/urllib/request.py", line 582, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python3.5/urllib/request.py", line 510, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.5/urllib/request.py", line 444, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.5/urllib/request.py", line 590, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

If there's anything else you would like to say (e.g. in case your issue is not about downloading a specific video; it might as well be a general discussion or proposal for a new feature), fill in the box below; otherwise, you may want to post an emoji or meme instead:

> Hello, my system is Ubuntu 16.04 LTS( Linux ubuntu 4.4.0-83-generic #106-Ubuntu SMP Mon Jun 26 17:54:43 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux ), my network provider is education network( I use an IPV6 proxy to download sources in IPV4).
> I checked my proxy by downloading video from other sources and it actually worked. So maybe some tricks get used by bilibili. Hope you could fix it. 
> Thanks a lot for providing such useful tool! :icecream:

汉语翻译最终日期：2016年02月26日

在提交前，请确保您已经检查了以下内容!

- [ ] 你可以在浏览器或移动端中观看视频，但不能使用`you-get`下载.
- [ ] 您的`you-get`为最新版.
- [ ] 我已经阅读并按 <https://github.com/soimort/you-get/wiki/FAQ> 中的指引进行了操作.
- [ ] 您的问题没有在<https://github.com/soimort/you-get/issues> , <https://github.com/soimort/you-get/wiki/FAQ> 或 <https://github.com/soimort/you-get/wiki/Known-Bugs> 报告，否则请在原有issue下报告.
- [ ] 本问题确实关于`you-get`, 而不是其他项目.

请使用`--debug`运行，并将输出粘贴在下面:

```
➜  ~ you-get 'http://www.bilibili.com/video/av11839449/' --debug -x 192.168.191.1:1080
[DEBUG] get_content: http://www.bilibili.com/video/av11839449/
[DEBUG] get_content: http://interface.bilibili.com/playurl?&cid=19551835&from=miniplay&player=1&sign=755fca43ed73826c9aae7a0fb4ba08de
[DEBUG] url_info: http://ws.acgvideo.com/8/37/19551835-1.flv?wsTime=1499169204&platform=pc&wsSecret2=9761c1664ee009c230e12eea8628a517&oi=1760342897&rate=6
you-get: version 0.4.750, a tiny downloader that scrapes the web.
you-get: ['http://www.bilibili.com/video/av11839449/']
Traceback (most recent call last):
  File "/usr/local/bin/you-get", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.5/dist-packages/you_get/__main__.py", line 92, in main
    main(**kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 1444, in main
    script_main('you-get', any_download, any_download_playlist, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 1355, in script_main
    download_main(download, download_playlist, args, playlist, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 1157, in download_main
    download(url, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 1437, in any_download
    m.download(url, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/extractors/bilibili.py", line 187, in bilibili_download
    info_only=info_only)
  File "/usr/local/lib/python3.5/dist-packages/you_get/extractors/bilibili.py", line 101, in bilibili_download_by_cid
    _, type_, temp = url_info(url, headers={'Referer': 'http://www.bilibili.com/'})
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 420, in url_info
    response = urlopen_with_retry(request.Request(url, headers=headers))
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 307, in urlopen_with_retry
    return request.urlopen(*args, **kwargs)
  File "/usr/lib/python3.5/urllib/request.py", line 163, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.5/urllib/request.py", line 472, in open
    response = meth(req, response)
  File "/usr/lib/python3.5/urllib/request.py", line 582, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python3.5/urllib/request.py", line 504, in error
    result = self._call_chain(*args)
  File "/usr/lib/python3.5/urllib/request.py", line 444, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.5/urllib/request.py", line 696, in http_error_302
    return self.parent.open(new, timeout=req.timeout)
  File "/usr/lib/python3.5/urllib/request.py", line 472, in open
    response = meth(req, response)
  File "/usr/lib/python3.5/urllib/request.py", line 582, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python3.5/urllib/request.py", line 510, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.5/urllib/request.py", line 444, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.5/urllib/request.py", line 590, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

如果您有其他附言，例如问题只在某个视频发生，或者是一般性讨论或者提出新功能，请在下面添加;或者您可以卖个萌:

> 下午好，我的电脑版本是Ubuntu 16.04 长期支持版( Linux ubuntu 4.4.0-83-generic #106-Ubuntu SMP Mon Jun 26 17:54:43 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux ), 我的网络是教育网(我用了IPV6的免流梯子去下载IPV4资源)
> 我使用`you-get`下载了别的源的资源文件，都能很好的运行，所以我只能推测是bilibili弄了新的反扒措施，希望开发者能搞定这个问题
> 最终灰常谢谢你们开发了这么好用的一个工具。:icecream:

**Comment by Hadesonly** [comment]
https://github.com/rosynirvana/you-get/tree/master

**Comment by kidozh** [comment]
I'll try, thx~

**Comment by rosynirvana** [comment]
the master branch of my fork is unstable
you can try #1698 if you need bilibili patch only

**Comment by jyuan15** [comment]
仍然不能解析和下载bilibili电影，bilibili其他视频可以解析

C:\Users\Administrator>you-get -i http://bangumi.bilibili.com/movie/12004
you-get: [error] oops, something went wrong.
you-get: don't panic, c'est la vie. please try the following steps:
you-get:   (1) Rule out any network problem.
you-get:   (2) Make sure you-get is up-to-date.
you-get:   (3) Check if the issue is already known, on
you-get:         https://github.com/soimort/you-get/wiki/Known-Bugs
you-get:         https://github.com/soimort/you-get/issues
you-get:   (4) Run the command with '--debug' option,
you-get:       and report this issue with the full output.

[DEBUG] url_locations: http://bangumi.bilibili.com/movie/12004
[DEBUG] get_content: http://bangumi.bilibili.com/movie/12004
[DEBUG] get_content: http://bangumi.bilibili.com/jsonp/seasoninfo/12004.ver?call
back=seasonListCallback&jsonp=jsonp&_=1499325574195
you-get: version 0.4.750-testing, a tiny downloader that scrapes the web.
you-get: ['http://bangumi.bilibili.com/movie/12004']
Traceback (most recent call last):
  File "C:\Python34\Scripts\you-get-script.py", line 9, in <module>
    load_entry_point('you-get==0.4.750', 'console_scripts', 'you-get')()
  File "C:\Python34\lib\site-packages\you_get\__main__.py", line 92, in main
    main(**kwargs)
  File "C:\Python34\lib\site-packages\you_get\common.py", line 1404, in main
    script_main('you-get', any_download, any_download_playlist, **kwargs)
  File "C:\Python34\lib\site-packages\you_get\common.py", line 1282, in script_m
ain
    download_main(download, download_playlist, args, playlist, output_dir=output
_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption
)
  File "C:\Python34\lib\site-packages\you_get\common.py", line 1066, in download
_main
    download(url, **kwargs)
  File "C:\Python34\lib\site-packages\you_get\common.py", line 1392, in any_down
load
    m.download(url, **kwargs)
  File "C:\Python34\lib\site-packages\you_get\extractor.py", line 73, in downloa
d_by_url
    self.prepare(**kwargs)
  File "C:\Python34\lib\site-packages\you_get\extractors\bilibili.py", line 127,
 in prepare
    self.bangumi_entry(**kwargs)
  File "C:\Python34\lib\site-packages\you_get\extractors\bilibili.py", line 181,
 in bangumi_entry
    ep_ids = collect_bangumi_epids(bangumi_data)
  File "C:\Python34\lib\site-packages\you_get\extractors\bilibili.py", line 249,
 in collect_bangumi_epids
    eps = json_data['result']['episodes']
KeyError: 'result'

**Comment by soimort** [comment]
Closed via #1698.
